README.md

```diff
@@ -1,8 +1,408 @@
+with `Generated(PropertyGeneration.Always)`.
+
+Example:
+
+```csharp
+Property(x => x.FullName, m =>
+{
+    m.Generated(PropertyGeneration.Always);
+});
+```
+
+Note that generated columns will be read back by NHibernate during any transaction that inserts or updates an entity
+that has at least one generated column. This has a couple of drawbacks:
+1. Any `INSERT`/`UPDATE` statement will automatically be followed by a `SELECT` statement to read back the generated value, even if the value is not needed by the application.
+2. Generated columns do not work with transactions that use Mutations instead of DML, as modifications that are executed as mutations cannot be read back during the transaction.
+
+One possible workaround for both the above is to define the column as a generated column in your database, but not
+mark it as generated in the NHibernate mapping. Instead the column should be marked as not-insertable and not-updatable
+to prevent NHibernate from writing a value to the column. The value will still be generated by Cloud Spanner, and if
+the generated value is needed by the application, a call to `session.Refresh(entity)` will read the value from the database.
+
+Example:
+
+```csharp
+// Mapping
+Property(x => x.FullName, m =>
+{
+    m.Insert(false);
+    m.Update(false);
+});
+
+// Usage
+var singer = new Singer { .. };
+session.Save(singer);
+// This will manually reload all properties from the database, including the generated FullName column.
+session.Refresh(singer);
+```
+
+## Mutations
+Cloud Spanner can use Mutations instead of DML to modify data. Mutations can be faster than DML, but Mutations do not support
+read-your-writes. It is recommended to read [https://cloud.google.com/spanner/docs/dml-versus-mutations](https://cloud.google.com/spanner/docs/dml-versus-mutations)
+if you are not familiar with the differences between Mutations and DML in Cloud Spanner.
+
+The Cloud Spanner NHibernate driver supports executing transactions that will use Mutations instead of DML. These transactions
+do not support read-your-writes. Any modifications will only be readable after the transaction has committed. Manually flushing
+the session during a transaction will also not send any modifications to Cloud Spanner.
+
+```csharp
+using var session = sessionFactory.OpenSession();
+using var transactionWithMutations = session.BeginTransaction(MutationUsage.Always);
+singer = session.Load<Singer>(singerId);
+singer.LastName = "Yates-Fish";
+
+// Committing the transaction will send the Mutation to Cloud Spanner and the updated value should be visible.
+transactionWithMutations.Commit();
+singersWithLastNameYatesFish = session.Query<Singer>().Where(s => s.LastName.Equals("Yates-Fish")).ToList();
+```
+
+# Limitations
+
+## Generated Values for Primary Keys
+Cloud Spanner does not support sequences, identity columns, or other value generators in the database that will
```

Comment:
Might be worth pointing to the docs where we recommend that you don't generate sequential PKs as an explanation for why we're asking to use guids.

Reply:
Done